obonet/__init__.py

```diff
@@ -9,12 +10,20 @@
 
 def _get_version() -> str | None:
     # https://github.com/pypa/setuptools_scm#retrieving-package-version-at-runtime
-    from pkg_resources import DistributionNotFound, get_distribution
+    if sys.version_info >= (3, 8):
+        from importlib.metadata import PackageNotFoundError, version
 
-    try:
-        return str(get_distribution("obonet").version)
-    except DistributionNotFound:
-        return None
+        try:
+            return version("obonet")
+        except PackageNotFoundError:
+            return None
+    else:
+        from pkg_resources import DistributionNotFound, get_distribution
+
+        try:
+            return str(get_distribution("obonet").version)
+        except DistributionNotFound:
+            return None
```

Comment:
Thanks for the contribution!

What do you think about bumping to 3.8 at

https://github.com/dhimmel/obonet/blob/fec6a82e53d01338c85e77039a4dc05288f6ab99/pyproject.toml#L31

and then removing the `from pkg_resources import DistributionNotFound, get_distribution` approach altogether?

Reply:
3.7 end of life as of June 27, 2023 and pip should know to install an older release of obonet if using python 3.7

Reply:
That's fine with me, I tend to always err on the side of broader compatibility unless there is a meaningful feature being blocked by it, but I think in this case, this would not be a big issue.

Reply:
Cool. I'm happy to make the changes after merging this or you can. Let me know which you prefer!

Reply:
I made the changes, feel free to make further modifications if needed.